mustache.go

```diff
@@ -112,15 +124,20 @@ type sectionNode struct {
 	elems    []node
 }
 
-func (n *sectionNode) render(t *Template, w *writer, c ...interface{}) error {
+func (n *sectionNode) render(t *Template, w *writer, seenNodes map[node]struct{}, c ...interface{}) error {
+	if !maybeAddNode(seenNodes, n) {
+		return nodeAlreadyVisitedError(n.String())
+	}
```

Comment:
This pattern is common everywhere. Recommend:

if err := tryAddNode(seenNodes, n); err != nil {
   return err
}


The definition of tryAddNode is

```
func tryAddNode(seenNodes map[node]struct{}, newNode node) error {
	if _, has := seenNodes[newNode]; !has {
		seenNodes[newNode] = struct{}{}
		return nil
	}
	return fmt.Errorf("node %v has been already visited in the mustache node graph", node)
}
```

Reply:
Thank you for the suggestion 💯 Will make the tweaks 😄